mlir/tools/mlir-tblgen/AttrOrTypeFormatGen.cpp

```diff
@@ -851,16 +897,20 @@ void DefFormat::genCommaSeparatedPrinter(
 
 void DefFormat::genParamsPrinter(ParamsDirective *el, FmtContext &ctx,
                                  MethodBody &os) {
-  genCommaSeparatedPrinter(llvm::to_vector(el->getParams()), ctx, os,
-                           [&](ParameterElement *param) {});
+  SmallVector<FormatElement *> args = llvm::map_to_vector(
+      el->getElements(), [](ParameterElement *param) -> FormatElement * {
+        return reinterpret_cast<ParameterElement *>(param);
+      });
```

Comment:
I'm confused by the functor here, it's taking a ParameterElement* and then casting to a ParameterElement*?

Reply:
Yeah, this should cast to a `FormatElement` instead. Updated it.